### PR TITLE
[R-package] [ci] Reduce manual effort for version updates

### PR DESCRIPTION
**Pull request by jameslamb**
This is the first step towards #3283 . It does not automate creating any artifacts (https://github.com/microsoft/LightGBM/issues/3283#issuecomment-670830603, https://github.com/microsoft/LightGBM/pull/3285#issuecomment-670968342), but it removes some of the manual steps needed to update the R package when the version of LightGBM is updated.

In https://github.com/microsoft/LightGBM/pull/3071#issuecomment-670243598 I proposed adding a `Makefile` with a `make release` recipe, but I decided not to do that in this PR because I thought it would be very thin and therefore not worth adding another file to the repo.

As of this PR, the only change that will need to be made in a release PR is to regenerate `R-package/configure`, which can be done with this command if you have Docker available (documented [here](https://github.com/microsoft/LightGBM/blob/master/R-package/README.md#changing-the-cran-package)):

```shell
docker run \
    -v $(pwd):/opt/LightGBM \
    -t ubuntu:18.04 \
    /bin/bash -c "cd /opt/LightGBM && ./R-package/recreate-configure.sh"
``` 

I tested this with values `3.0.0`, `5.2.1` and `5.2.1rc` in `VERSION.txt`, just to make sure it all works as expected.

### CODE REVIEW

File: build_r.R [hunk 1]
```diff
@@ -155,6 +155,31 @@ result <- file.copy(
 )
 .handle_result(result)
 
+# R packages cannot have versions like 3.0.0rc1, but
+# 3.0.0-1 is acceptable
+LGB_VERSION <- readLines("VERSION.txt")[1]
```
Comment (guolinke) [inline build_r.R:160]:
```suggestion
LGB_VERSION <- readLines("VERSION.txt")[1L]
```

**Comment by guolinke** [comment]
@jameslamb I think this will be included in 3.0.0 ?

**Comment by jameslamb** [comment]
yes it should be!

Sorry, should have clarified...I built the package in https://github.com/microsoft/LightGBM/issues/3293#issuecomment-679431463 from this branch, so it is ok.

**Review by guolinke — APPROVED** [review]

**Comment by StrikerRUS** [comment]
@jameslamb 

> In #3071 (comment) I proposed adding a Makefile with a make release recipe, but I decided not to do that in this PR because I thought it would be very thin and therefore not worth adding another file to the repo.

Absolutely agree!

File: R-package/recreate-configure.sh [hunk 2]
```diff
@@ -4,6 +4,14 @@
 # this script should run on Ubuntu 18.04
 AUTOCONF_VERSION=$(cat R-package/AUTOCONF_UBUNTU_VERSION)
 
+# R packages cannot have versions like 3.0.0rc1, but
+# 3.0.0-1 is acceptable
+LGB_VERSION=$(cat VERSION.txt | sed "s/rc/-/g")
```
Comment (StrikerRUS) [inline R-package/recreate-configure.sh:9]: I'm confused a little bit. Where do we copy `VERSION.txt` from the root repo folder to `R-package`? Or how is it working?

**Review by StrikerRUS — APPROVED** [review]
Thanks, LGTM if it is working inside a Docker.

File: R-package/recreate-configure.sh (hunk 2 above)
Comment (jameslamb) [reply R-package/recreate-configure.sh:9]: you don't need to! When you run `./R-package/recreate-configure.sh`, the effective working directory is the root of the repo. So no need to copy it from anywhere.

Comment (StrikerRUS) [reply R-package/recreate-configure.sh:9]: Aha, got it! Thanks for the explanation!

**Comment by jameslamb** [comment]
> Thanks, LGTM if it is working inside a Docker.

thanks! I can confirm that it is. The diff in `configure` in this PR was generated with the `docker run` command in the README. Everything else in this PR will work outside of Docker, for example anywhere that people run `build-cran-package.sh` or `Rscript build_r.R`.

Now that this is automated, it will also fail builds where we update `VERSION.txt` but forget to update `R-package/configure`, which I think is a good thing!

**Pull request merged**